src/r2mm/profiles/ProfilesClient.ts

```diff
@@ -6,45 +6,60 @@ const getProfileUrl = (profileImportCode: string): string => {
     return `https://thunderstore.io/api/experimental/legacyprofile/get/${profileImportCode}/`;
 }
 
-async function createProfile(payload: string): Promise<AxiosResponse<{ key: string }>> {
-    try {
-        return await Axios.post(
-            "https://thunderstore.io/api/experimental/legacyprofile/create/",
-            payload,
-            {
-                headers: { 'Content-Type': 'application/octet-stream' }
-            }
-        );
-    } catch (e: any) {
-        if (Axios.isAxiosError(e) && e.response) {
-            if (e.response.status == 429) {
-                let message = e.message;
-                try {
-                    message = e.response.data.detail || e.message;
-                } catch {
-                }
-                throw new R2Error(
-                    "Too many exports in a short period of time",
-                    message,
-                    "Wait for a minute and try again"
-                );
-            } else {
-                throw new R2Error(
-                    "Failed to upload profile",
-                    e.message,
-                    `${e.response.status}`
-                );
+function formatApiError<T>(e: T, genericTitle: string): R2Error | T {
+    if (Axios.isAxiosError(e) && e.response) {
+        if (e.response.status == 429) {
+            let message = e.message;
+            try {
+                message = e.response.data.detail || e.message;
+            } catch {
             }
+            return new R2Error(
+                "Too many attempts in a short period of time",
+                message,
+                "You were rate limited by the server, wait for a while and try again."
+            );
+        } else {
+            return new R2Error(
+                genericTitle,
+                e.message,
+                `${e.response.status}`
+            );
+        }
+    }
+    return e;
+}
+
+async function handleApiErrors<I, J>(
+    apiCall: Promise<AxiosResponse<I, J>>,
+    errorTitle: string,
+): Promise<AxiosResponse<I, J>> {
+    try {
+        const response = await apiCall;
+        if (Axios.isAxiosError(response)) {
+            throw response;
         }
-        throw e;
+        return response;
+    } catch (e: any) {
+        throw formatApiError(e, errorTitle);
     }
 }
 
-async function getProfile(profileImportCode: string): Promise<AxiosResponse<string>> {
+function createProfile(payload: string): Promise<AxiosResponse<{ key: string }>> {
```

Comment:
A bit confusing how you drop the async/await from this middle function and I don't quite know if it has any benefits given how it's currently used?

Reply:
async + await is slightly less efficient than simply returning a promise from a sync function is (due to the overhead caused by context swaps, especially if there are _a lot_ of other async code pending in the queue), in some niche cases significantly so. I'd not say it's necessarily a bad practice to use async + await here, but personally I'm in the habit of not using it unless necessary